### PR TITLE
Support use /dev/stdin as SOURCE path for image build

### PR DESCRIPTION
**Pull request by zyfjeff**
## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

## Details

Support use /proc/self/fd/0 as SOURCE path for image build

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/bin/nydus-image/main.rs [hunk 1]
```diff
@@ -1804,3 +1805,12 @@ impl Command {
         nydus_service::validate_threads_configuration(v).map(|s| s.to_string())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::Command;
+    #[test]
+    fn test_ensure_file() {
+        Command::ensure_file("/proc/self/fd/0").unwrap();
```
Comment (imeoer) [inline src/bin/nydus-image/main.rs:1814]: Maybe use `/dev/stdin`? It seems more compatible with macos. (the unit tests will run in macos env in release action).

Comment (zyfjeff) [reply src/bin/nydus-image/main.rs]: fixed

Comment (zyfjeff) [reply src/bin/nydus-image/main.rs]: https://github.com/zyfjeff/image-service/actions/runs/5997383931/job/16263668460

release action ci ran successfully 

ping @imeoer

Comment (imeoer) [reply src/bin/nydus-image/main.rs]: Thanks for the testing!

File: src/bin/nydus-image/main.rs [hunk 2]
```diff
@@ -1695,9 +1695,10 @@ impl Command {
         let file_type = metadata(path.as_ref())
             .context(format!("failed to access path {:?}", path.as_ref()))?
             .file_type();
+        // support /proc/self/fd/0 as input for SOURCE, it's a char device
```
[the remark this answers is not in the document]
Comment (zyfjeff) [reply src/bin/nydus-image/main.rs:1698]: fixed

**Pull request closed without merging**